libs/utils/src/structural_sharing.ts

```diff
@@ -0,0 +1,21 @@
+import { deepEqual } from '@votingworks/basics';
+
+/**
+ * React Query allows passing a custom `structuralSharing` function to `useQuery`.
+ * It allows us to compare the old and new data and keep the old data as the
+ * query data if it is deep equal to the new data. As a result, we can maintain
+ * the reference to the old data and prevent React from re-rendering components
+ * that depend on it.
+ */
+export function persistDataReferenceIfDeepEqual<T>(
+  oldData: T | undefined,
+  newData: T
+): T {
+  if (!oldData) {
+    return newData;
+  }
+
+  // Prevent unnecessary re-renders of dependent components
+  const isUnchanged = deepEqual(oldData, newData);
+  return isUnchanged ? oldData : newData;
+}
```

Comment:
Here's the function we're replacing: https://github.com/TanStack/query/blob/1fc6124bfd22a26095ab1fca20caa39b4036fd86/packages/query-core/src/utils.ts#L244 (call site: https://github.com/TanStack/query/blob/1fc6124bfd22a26095ab1fca20caa39b4036fd86/packages/query-core/src/utils.ts#L357)

Maybe you could test out running that function on some of our returned data to figure out why this isn't working out of the box? (Also, I didn't track down the exact version of react-query that we're using, so it might be worth doing that before testing)

Reply:
Thank you for the pointers! I followed the code and found that their default function only considers objects equal if they both have the plain object prototype. This means that our `DateWithoutTime` and even the inbuilt `Date` objects were causing equality checks to fail.